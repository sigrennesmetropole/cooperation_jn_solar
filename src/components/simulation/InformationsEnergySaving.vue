<script setup lang="ts">
import energySaving1 from '@/assets/illustrations/energy_saving_1.svg'
import energySaving2 from '@/assets/illustrations/energy_saving_2.svg'
import energySaving3 from '@/assets/illustrations/energy_saving_3.svg'
import { ref } from 'vue'

const informations = [
  {
    id: 1,
    img: energySaving1,
    title: 'Qu’est-ce que l’autoconsommation ?',
    description:
      'Consommer chez vous l’électricité que vous avez produite et vendre le surplus sur le réseau public, c’est ce qu’on appelle l’autoconsommation.',
  },
  {
    id: 2,
    img: energySaving2,
    title: 'Maitriser au mieux votre budget électricité.',
    description:
      'En devenant producteur d’une partie de votre électricité,  votre budget énergie est moins dépendant des évolutions de prix de votre fournisseur.',
  },
  {
    id: 3,
    img: energySaving3,
    title: 'Pour aller plus loin',
    description:
      "Vous pourrez économiser davantage d'énergie si vos appareils électriques fonctionnent au moment où les panneaux solaires produisent, c’est-à-dire au cours de la journée.",
  },
]
const currentInfo = ref(1)
</script>

<template>
  <div class="rounded-lg bg-slate-100 h-[402px] w-full px-6 py-8">
    <template v-for="info in informations" :key="info.id">
      <div
        v-if="info.id === currentInfo"
        class="flex flex-col justify-center items-center"
      >
        <img class="w-[354px] h-[160px]" :src="info.img" alt="" />
        <span class="font-dm-sans font-bold text-base mt-4">
          {{ info.title }}
        </span>
        <span
          class="font-dm-sans font-normal text-sm mt-2 text-center h-[80px]"
        >
          {{ info.description }}
        </span>
      </div>
    </template>
    <div class="flex flex-row gap-3 mt-4 justify-center items-center">
      <div
        v-for="info in informations"
        tabindex="0"
        :key="info.id"
        class="w-6 h-6 rounded-full flex justify-center items-center cursor-pointer"
        :class="
          info.id === currentInfo ? 'bg-slate-600' : 'border border-slate-300'
        "
        @click="currentInfo = info.id"
        @keydown.enter="currentInfo = info.id"
      >
        <span
          class="font-dm-sans font-bold text-xs"
          :class="info.id === currentInfo ? 'text-white' : 'text-slate-500'"
        >
          {{ info.id }}
        </span>
      </div>
    </div>
  </div>
</template>
