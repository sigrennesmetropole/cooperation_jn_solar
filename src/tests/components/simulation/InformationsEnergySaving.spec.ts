import InformationsEnergySaving from '@/components/simulation/InformationsEnergySaving.vue'
import { describe, it, expect, beforeEach } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'

describe('InformationsEnergySaving.vue', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(InformationsEnergySaving)
  })

  it('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays the correct information for the currentInfo', () => {
    const title = wrapper.find('span.font-bold')
    const description = wrapper.find('span.font-normal')
    expect(title.text()).toBe('Qu’est-ce que l’autoconsommation ?')
    expect(description.text()).toContain('Consommer chez vous')
  })

  it('changes currentInfo when a different circle is clicked', async () => {
    const secondCircle = wrapper.findAll('.cursor-pointer')[1]
    await secondCircle.trigger('click')
    const title = wrapper.find('span.font-bold')
    const description = wrapper.find('span.font-normal')
    expect(title.text()).toBe('Maitriser au mieux votre budget électricité.')
    expect(description.text()).toContain('En devenant producteur')
  })
})
