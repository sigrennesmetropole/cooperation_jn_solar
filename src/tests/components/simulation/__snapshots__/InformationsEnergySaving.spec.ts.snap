// Vitest Snapshot v1

exports[`InformationsEnergySaving.vue > renders the component correctly 1`] = `
"<div class=\\"rounded-lg bg-slate-100 h-[402px] w-full px-6 py-8\\">
  <div class=\\"flex flex-col justify-center items-center\\"><img class=\\"w-[354px] h-[160px]\\" src=\\"/src/assets/illustrations/energy_saving_1.svg\\" alt=\\"\\"><span class=\\"font-dm-sans font-bold text-base mt-4\\">Qu’est-ce que l’autoconsommation ?</span><span class=\\"font-dm-sans font-normal text-sm mt-2 text-center h-[80px]\\">Consommer chez vous l’électricité que vous avez produite et vendre le surplus sur le réseau public, c’est ce qu’on appelle l’autoconsommation.</span></div>
  <!--v-if-->
  <!--v-if-->
  <div class=\\"flex flex-row gap-3 mt-4 justify-center items-center\\">
    <div tabindex=\\"0\\" class=\\"w-6 h-6 rounded-full flex justify-center items-center cursor-pointer bg-slate-600\\"><span class=\\"font-dm-sans font-bold text-xs text-white\\">1</span></div>
    <div tabindex=\\"0\\" class=\\"w-6 h-6 rounded-full flex justify-center items-center cursor-pointer border border-slate-300\\"><span class=\\"font-dm-sans font-bold text-xs text-slate-500\\">2</span></div>
    <div tabindex=\\"0\\" class=\\"w-6 h-6 rounded-full flex justify-center items-center cursor-pointer border border-slate-300\\"><span class=\\"font-dm-sans font-bold text-xs text-slate-500\\">3</span></div>
  </div>
</div>"
`;
